### ISSUE TITLE
Feature Request: Offline Spending

### ISSUE THREAD

**Issue by jenniferberger** [m2049r/xmrwallet#125]
As a paranoid user i want to spend and receive monero without internet.

Without a sim-card it is impossible to exchange monero in public places. There are applications that require the user to turn of network to avoid tracking etc. Therefore the application should be able to exchange a signed transaction with QR-codes.
From my understanding this should be possible with the submit_transfer, sign_transfer methods of monero-wallet-cli. Not sure what happens if a signed transaction is broadcasted twice.

Sender:
In the last step of sendfragment add a additional button 'Show Signed Transaction' that displays the data as QR-code. Below the QR-code a 'mark as spend until online' button should be visible. Pressing this button displays the transaction in the UI as if it was sent to the network, maybe with a different icon.
The app needs to notify the sender in the LoginActivity that there are stored, unsent transactions available.
Unsent transactions need a list, item and a broadcast now button.

Receiver:
In the receive dialog a button 'Scan Signed Transaction' opens the QR-scanner. The QR-scanner returns with the data of the Signed Transaction. The dialog then displays the details of the transaction (like payment id, maybe the receiver address if it is for the current wallet).
The receiver can now decide to store the transaction (make him trust the sender not to double spend), broadcast the transaction (can even broadcast for other receivers).
The app needs to notify the receiver in the LoginActivity that there are stored, unsent transactions available.
Unsent transactions need a list, item and a broadcast now button.

The receiver has a high interest to broadcast the transaction in order to get the funds. There should be a warning when both receiver and sender use this in offline mode.

The receiver code should be careful interpreting the transaction from the QR-code, check lengths/value ranges...

**Comment by m2049r** [comment]
wow - this is something :) need to put thought into this, and i don't see it coming in the immediate future.

one question: how do you prevent the sender turning her phone on and sweep_all her XMR to another wallet immediately following the physical transaction described above?

**Comment by jenniferberger** [comment]
It is not possible to completely prevent that the sender scams the receiver. Ideally the receiver has internet before the sender.
The warning message on the 'scan signed transaction' activity  should highlight the possibilities that could cause the transaction to get invalid.

**Comment by m2049r** [comment]
can multisig help in this scenario?

**Comment by jenniferberger** [comment]
i think more of the sender is my fingerprint locked, stripped down single-wifi android that exports the transaction to the lte-connected malware contaminated spy device that should not have my wallet.

multisig is scratching the issue as this requires to pass a signed transaction to the other party before broadcasting. Maybe the requested feature is a multisig support that allows sharing via QR and broadcasting even if the wallet is not multisig.

**Comment by lessless** [comment]
that's absolutely neat

**Comment by golere** [comment]
This also would enable offline wallets or selfmade hardware wallets:

You run a wallet on a offline device (f.e. Raspberry Pi) which is never connected to the internet. When spending Monero, you just sign the transaction on the offline device, Scan the signed tx and transfer it by QR-code(s) to monerujo to broadcast it to the network.

Also see https://github.com/xloem/qrstream and #144

**Comment by lessless** [comment]
Exactly! All kinds of interesting scenarios with using smartphone as a cold storage.
